packages/react/src/react/views/collections/list.client.tsx

```diff
@@ -189,43 +323,66 @@ export function ListTable(props: ListTableProps) {
       setPagination({
         page: newPagination.pageIndex + 1,
         pageSize: newPagination.pageSize,
+        search: pagination.search,
+        sortBy: pagination.sortBy,
+        sortOrder: pagination.sortOrder,
       })
     },
     getPaginationRowModel: getPaginationRowModel(),
     state: {
       rowSelection,
-      sorting,
+      sorting: initialSorting,
```

Comment:
The state would be `{ id: pagination.sortBy, desc: pagination.sortOrder === "desc" }`